Lib/socketserver.py

```diff
@@ -652,6 +652,11 @@ def process_request_thread(self, request, client_address):
             self.handle_error(request, client_address)
         finally:
             self.shutdown_request(request)
+            t = threading.current_thread()
```

Comment:
You need a lock to protect self._threads. You should not call .remove() if t.daemon is true.

Please rename t to thread.

We should be extra careful and use a nested finally block in case shutdown_request() raises a new exception:

```
finally:
  try:
    self.shutdown_request(request)
  finally:
    ...
```

Reply:
thank for your reviewing.
I fixed and committed points you suggested.
Could you re-check this?